bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudAccountHandler.java

```diff
@@ -155,6 +156,36 @@ public DeviceStatus fetchDeviceStatus(int deviceId, Optional<Integer> buildingId
         }
     }
 
+    public HeatpumpDeviceStatus sendHeatpumpDeviceStatus(HeatpumpDeviceStatus heatpumpDeviceStatus)
+            throws MelCloudCommException, MelCloudLoginException {
+        connectIfNotConnected();
+        try {
+            return connection.sendHeatpumpDeviceStatus(heatpumpDeviceStatus);
+        } catch (MelCloudLoginException e) {
+            throw e;
+        } catch (MelCloudCommException e) {
+            logger.debug("Sending failed, retry once with relogin");
+            connect();
+            return connection.sendHeatpumpDeviceStatus(heatpumpDeviceStatus);
+        }
+    }
+
+    public HeatpumpDeviceStatus fetchHeatpumpDeviceStatus(int deviceId, Optional<Integer> buildingId)
+            throws MelCloudCommException, MelCloudLoginException {
+        connectIfNotConnected();
+        int bid = buildingId.orElse(findBuildingId(deviceId));
+
+        try {
+            return connection.fetchHeatpumpDeviceStatus(deviceId, bid);
+        } catch (MelCloudLoginException e) {
+            throw e;
```

Comment:
This looks strange, but is because `MelCloudLoginException` extends `MelCloudCommException`. However, it adds nothing to add `MelCloudLoginException` as throws to this method. Maybe it's better to have `MelCloudLoginException` just make it's own exception and not a extends of `MelCloudCommException`.

Reply:
Thanks for checking.
I guess for now it would be enough to have `MelCloudLoginException` extend `Exception` instead of `MelCloudCommException`?

Reply:
I'll create a new PR for the `MelCloudLoginException` shortly.